install: add note regarding arm macos support (#3692)
iterative/dvc.org: content/docs/install/macos.md

# Installation on macOS

<admon type="tip">

To use DVC [as a Python library](/doc/api-reference), please

> [install with pip](#install-with-pip) or [with conda](#install-with-conda).

Reviewer: ```suggestion
install [with `pip`](#install-with-pip) or [with `conda`](#install-with-conda).
```

> </admon>

Reviewer: - [ ] We should apply this to all installation pages...

> > To use DVC [as a Python library](/doc/api-reference), please
> > [install with pip](#install-with-pip) or [with conda](#install-with-conda).

Reviewer: ```suggestion
<admon type="tip">

To use DVC [as a Python library](/doc/api-reference), please
[install with pip](#install-with-pip) or [with conda](#install-with-conda).

</admon>
```

> > ⚠️ To use DVC on ARM (Apple Silicon) macOS, you must
> > [install with homebrew](#install-with-homebrew) or
> > [with pip](#install-with-pip). The PKG binary and conda installation methods
> > can only be used on x86 (Intel) macOS and are not supported for ARM macOS.

Reviewer: ```suggestion
<admon type="warn">

To use DVC on ARM (Apple Silicon) macOS, you must install
[with homebrew](#install-with-homebrew) or [with `pip`](#install-with-pip).
The PKG binary and `conda` installation methods can only be used on x86 (Intel)
macOS.

</admon>
```
